Improve transcript widget edit location actions

shashankbrgowda opened the pull request:
Fixes transcript widget location editing

garrettjstevens commented on packages/jbrowse-plugin-apollo/src/FeatureDetailsWidget/TranscriptWidgetEditLocation.tsx:
```diff
@@ -145,11 +150,13 @@ export const TranscriptWidgetEditLocation = observer(
     }
 
-    function handleExonLocationChange(
+    async function handleExonLocationChange(
```
Instead of making async, use `void changeManager.submit()`

garrettjstevens commented on packages/jbrowse-plugin-apollo/src/FeatureDetailsWidget/TranscriptWidgetEditLocation.tsx:
```diff
@@ -159,31 +166,146 @@ export const TranscriptWidgetEditLocation = observer(
+          )
+
+          // Submit all changes sequentially. gene -> transcript -> exon
+          for (const change of changes) {
```
These should all be in a `changes` attribute of a single `LocationStartChange` instead of three different `LocationStartChange`s. So it would look something like this:

```js
new LocationStartChange({
  typeName: 'LocationStartChange',
  changedIds: ['feature1Id', 'feature2Id'],
  assembly,
  changes: [
    { featureId: 'feature1Id', oldStart: 500, newStart: 490 },
    { featureId: 'feature2Id', oldStart: 500, newStart: 490 },
  ],
})
```

Same for the `LocationEndChange`s.

garrettjstevens commented:
The logic for whether or not something was in a gene glyph being in `mouseEvents` makes it so non-gene features can't be dragged anymore. I've proposed an alternate solution in a branch here that makes the parent/child expansion more general and has the gene glyph-specific logic outside of `mouseEvents`: [555_click_drag_improvement](https://github.com/GMOD/Apollo3/tree/555_click_drag_improvement).

garrettjstevens commented:
I've split this into two PRs, this one for the transcript widget and another for click and drag actions

garrettjstevens commented:
I was testing with a feature that looks like this:

![image](https://github.com/user-attachments/assets/71eb5d6c-1340-482c-9486-c9b9ac6d5300)

And when I changed the last exon's max to something too small (e.g. I changed 19200 to 100), it got stuck in a loop of submitting changes for that position.

shashankbrgowda commented:
> I was testing with a feature that looks like this:
> 
> ![image](https://private-user-images.githubusercontent.com/25592344/446789356-71eb5d6c-1340-482c-9486-c9b9ac6d5300.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDc5NzgwMDYsIm5iZiI6MTc0Nzk3NzcwNiwicGF0aCI6Ii8yNTU5MjM0NC80NDY3ODkzNTYtNzFlYjVkNmMtMTM0MC00ODJjLTk0ODYtYzliOWFjNmQ1MzAwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTIzVDA1MjE0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM5YzE2N2E3ZDE5OWQxODUwNTQzMmUyZGQyM2QzMTc2MjFjNjk5M2IwMTAyMGIwM2JkZDc3MDVlM2VmYTRhYmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.NPqmkjM96xNwQD-DVdKxuNF2L8QmMdV2NCNMIJJAuvs)
> 
> And when I changed the last exon's max to something too small (e.g. I changed 19200 to 100), it got stuck in a loop of submitting changes for that position.

Thanks @garrettjstevens 

This should be fixed. Haven't pushed my changes yet
